tests/nodeos_read_terminate_at_block_test.py

```diff
@@ -0,0 +1,316 @@
+                 lib = int(info["last_irreversible_block_num"])
+                 forkDbHead =  int(info["fork_db_head_block_num"])
+                 Utils.Print("head=" + str(head)  + ", lib=" + str(lib) + ", terminate-at-block=", str(termAtBlockNum))
+                 Utils.Print(info)
+             else:
+                 break
+         assert head == lib, "Head ({}) should be equal to lib ({})".format(head, lib)
+         assert head >= termAtBlockNum, "Head ({}) should no less than termAtBlockNum ({})".format(head, termAtBlockNum)
+         assert forkDbHead >= head, "Fork db head ({}) should be larger or equal to the head ({})".format(forkDbHead, head)
+      finally:
+         stopProdNode()
+         Utils.Print("End replayInIrrModeWithRevBlksAndConnectedToProdNodeTermAtBlock")
+
+   # 2th test case: Replay in speculative mode with reversible blks while connected to producing node
+   # Expectation: Node replays and launches successfully
+   #              and the head and lib should be no less than termAtBlockNum
+   def replayInSpecModeWithRevBlksAndConnectedToProdNodeTermAtBlock(nodeIdOfNodeToTest, nodeToTest):
+      try:
+         startProdNode()
+         # Kill node and replay in speculative  mode
+         nodeToTest.kill(signal.SIGTERM)
+         waitForBlksProducedAndLibAdvanced() # Wait
+         infoprod = producingNode.getInfo()
+         headprod = int(infoprod["head_block_num"])
+         termAtBlockNum = headprod + termAtFutureBlockNum
+        
+         relaunchTestNode(nodeToTest, chainArg=" --read-mode speculative  --terminate-at-block=" + str(termAtBlockNum))
+         Utils.Print("Test node begin recevie from producing node and is expected to stop at or little bigger than block number " + str(termAtBlockNum) )
+
+         while True :
+             info = nodeToTest.getInfo()
+             if info is not None:
+                 head = int(info["head_block_num"])
+                 lib = int(info["last_irreversible_block_num"])
+                 forkDbHead =  int(info["fork_db_head_block_num"])
+                 Utils.Print("head=" + str(head)  + ", lib=" + str(lib) + ", terminate-at-block=", str(termAtBlockNum))
+                 Utils.Print(info)
+             else:
+                 break
+         assert head >= termAtBlockNum, "Head ({}) should be no less than termAtBlockNum ({})".format(head, termAtBlockNum)
+         assert forkDbHead >= head, "Fork db head ({}) should be larger or equal to the head ({})".format(forkDbHead, head)
+      finally:
+         stopProdNode()
+         Utils.Print("End replayInSpecModeWithRevBlksAndConnectedToProdNodeTermAtBlock")
+
+   # 3th test case: Replay in head mode with reversible blks while connected to producing node
+   # Expectation: Node replays and launches successfully
+   #              and the head and lib should be no less than termAtBlockNum
+   def replayInHeadModeWithRevBlksAndConnectedToProdNodeTermAtBlock(nodeIdOfNodeToTest, nodeToTest):
```

Comment:
same comments apply as for:
`replayInIrrModeWithRevBlksAndConnectedToProdNodeTermAtBlock`